protostar/utils/starknet_compilation.py

```diff
@@ -57,12 +58,15 @@ def build_codes(*cairo_file_paths: Path) -> List[Tuple[str, str]]:
 
     def preprocess_contract(
         self, *cairo_file_paths: Path
-    ) -> StarknetPreprocessedProgram:
+    ) -> Union[StarknetPreprocessedProgram, TestCollectorPreprocessedProgram]:
```

Comment:
It seems kinda counterintuitive that the `compiler` class does both compilation and/or test collection. And it's kinda bad that this broad scope leaks in types.

Would it make sense to move test collection outside this class? It seems like the building blocks are all static here so this should be easily doable.

Reply:
I have a separate task to refactor this StarknetCompiler class because it became a mess. I want to remove stuff releated to `fast-collection` first though.

Reply:
I want to keep size of the PR managable, can we move this request to #485 ?